components/compositing/compositor.rs

```diff
@@ -1957,17 +1957,15 @@ impl<Window: WindowMethods> IOCompositor<Window> {
         let clip_rect_for_children = if masks_to_bounds {
             Rect::new(Point2D::zero(), clipped_layer_bounds.size)
         } else {
-            clipped_layer_bounds.translate(&clip_rect.origin)
+            clipped_layer_bounds.translate(&-layer_bounds.origin)
         };
 
         let child_point = point - layer_bounds.origin;
         for child in layer.children().iter().rev() {
             // Translate the clip rect into the child's coordinate system.
-            let clip_rect_for_child =
```

Comment:
Yes, it is. I'll go back to the drawing board and make sure the fix works when the parent and/or child documents are scrolled.